### PR TITLE
[FIX] hr_holidays_public: Consider public holdays for leaves in hours

### PR DESCRIPTION
**Pull request by maisim**
Backport of

7d5bc92 [FIX] hr_holidays_public: Exclude public holidays on hours computation on validated leave
887fd4a6 [FIX] hr_holidays_public: Being multi-record aware in _compute_number_of_hours_display

from 13.0 ( hr-holidays repo )

Make it working on 12.0

Refactor tests files: 

- Move to TransactionCase
- Split  test file :
  - one common file (the common setUp) 
  - one file for tests on holidays in days
  - one file in holidays in hours

### CODE REVIEW

File: hr_holidays_public/__manifest__.py
Comment (pedrobaeza) [inline hr_holidays_public/__manifest__.py]: Please reduce the diff to the minimal needed.

File: hr_holidays_public/models/hr_leave.py
Comment (maisim) [inline hr_holidays_public/models/hr_leave.py]: @pedrobaeza 

Did you remember why you filter on state ? It make the displayed value in the form false (when we create a new leave)

You can check here (today is public holiday ! :p )
http://oca-hr-holidays-13-0-d4d9fdf82183.runboat.odoo-community.org/web#action=168&model=hr.leave&view_type=list&cids=&menu_id=117

When I remove this filter it's ok but I cannot undestand what it is for

Comment (pedrobaeza) [reply hr_holidays_public/models/hr_leave.py]: The explanation is in the method docstring.

Comment (maisim) [reply hr_holidays_public/models/hr_leave.py]: I saw, but has I understood how it work we always need to push exclude_public_holidays=True in the context to get the good number of hours whatever the state

Comment (pedrobaeza) [reply hr_holidays_public/models/hr_leave.py]: Not on v13 at least, and not if the holidays type doesn't have the check.

Comment (maisim) [reply hr_holidays_public/models/hr_leave.py]: Today is defined as public holiday
image

"Compensatory days" type exclude public holidays
image

This is not taken into account in the form view
image

From http://oca-hr-holidays-13-0-d4d9fdf82183.runboat.odoo-community.org/web#action=168&model=hr.leave&view_type=list&cids=&menu_id=117

I have the same behavor on 12.0. Remove state filter fix the problem

I'm really sorry if there's something I don't understand and I'm wasting your time but I feel like it's not working the way it should

Comment (pedrobaeza) [reply hr_holidays_public/models/hr_leave.py]: OK, then remove it such condition. Have you seen if in v13 is happening the same problem?

Comment (maisim) [reply hr_holidays_public/models/hr_leave.py]: Ok it can't work as is on a vanilla 12.0

Let me take a closer look, I will let you know when the PR is ready.

For 13.0, if you confirm the problem I can open an issue or submit a PR

Comment (pedrobaeza) [reply hr_holidays_public/models/hr_leave.py]: That's the question. I don't think there's an issue in v13. The problem is here on v12.

Comment (maisim) [reply hr_holidays_public/models/hr_leave.py]: The screenshots are from the link above (13.0 runboat instance)

**Comment by maisim** [comment]
Hi @pedrobaeza ,
Can you please have a look on it ? 
I don't know what is the good way to get all green as the only solution a can see is to add tests to not concerned by this PR parts of code to improve coverage
Thanks in advance for your help

**Review by pedrobaeza — APPROVED** [review]

**Review by nimarosa — CHANGES_REQUESTED** [review]
Don't change version number in manifest. The rest LGTM.

**Comment by maisim** [comment]
> Don't change version number in manifest. The rest LGTM.

Hi @nimarosa, Thanks for your review.
Just to understand, can you explain why the version number should not change ?
This PR is a bugfix and will change the behavior of the plugin (requests in hours should work correcty), In my mind it is a case where the version number have to change.

If I am wrong, Is https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#version-numbers need an update?

**Comment by mtelahun** [comment]
The version numbers will be automatically bumped by ocabot 
(https://github.com/OCA/oca-github-bot) when your PR is merged.

**Comment by maisim** [comment]
ping @OCA/human-resources-maintainers

Any chance to get it merged ? Let me know if I can do something to help

**Comment by SirTakobi** [comment]
@OCA/human-resources-maintainers ^

**Comment by nimarosa** [comment]
Let's try again. Codecov should not block the merge. 

/ocabot merge patch

**Review by nimarosa — APPROVED** [review]

**Pull request merged**
